packages/core/src/codewhisperer/tracker/lineTracker.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { isTextEditor } from '../../shared/utilities/editorUtilities'
+
+export interface LineSelection {
+    anchor: number
+    active: number
+}
+
+export interface LinesChangeEvent {
+    readonly editor: vscode.TextEditor | undefined
+    readonly selections: LineSelection[] | undefined
+
+    readonly reason: 'editor' | 'selection' | 'content'
+}
+
+/**
+ * This class providees a single entry point interface to manage and access users' "line" selections
+ * Callers could use it by subscribing onDidChangeActiveLines to do UI updates or logic needed to be executed when line selections get changed
+ */
+export class LineTracker implements vscode.Disposable {
```

Comment:
why is this not part of `CodeWhispererTracker` ? should `CodeWhispererTracker` be renamed to `ModificationsTracker` ?

Reply:
> should CodeWhispererTracker be renamed to ModificationsTracker ?

yea it should be renamed to `ModificationTracker`.

Reply:
> why is this not part of CodeWhispererTracker

I think it makes more sense `LineTracker` should combine with `CodeCoverageTracker` as they both 2 are subscribing DocumentChangedEvent, however I would prefer extracting this part to another later `DocumentChangeTracker` given the fact that "LineTracker" sounds more reasonable to only fire events when there is "line" change but not content change.

Reply:
> I think it makes more sense `LineTracker` should combine with `CodeCoverageTracker` as they both 2 are subscribing DocumentChangedEvent, however I would prefer extracting this part to another component maybe `DocumentChangeTracker`.

ok. Hoping to see a followup PR for that.